src/useVisibilityObserver/useVisibilityObserver.ts

```diff
@@ -0,0 +1,24 @@
+import { onMounted, ref, Ref } from 'vue'
+import { useIntersectionObserver } from '@/useIntersectionObserver'
+
+export function useVisibilityObserver(element: Ref<HTMLElement | undefined>): Ref<boolean> {
+  const visible = ref(false)
+
+  function intersect(entries: IntersectionObserverEntry[]): void {
+    entries.forEach(entry => {
+      if (entry.isIntersecting) {
+        visible.value = true
+        disconnect()
+      }
```

Comment:
As a generic composition I think we shouldn't auto disconnect and we should track visibility. 

```suggestion
      visible.value = entry.isIntersecting
```

Reply:
Since its a common use case though maybe we can add a `disconnectWhenVisible: boolean` option that does do the disconnect automatically.